pygments/lexers/tal.py

```diff
@@ -0,0 +1,68 @@
+"""
+    pygments.lexers.tal
+    ~~~~~~~~~~~~~~~~~~~
+
+    Lexers for Uxntal
+
+    .. versionadded:: 2.12
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import re
+import keyword
+
+from pygments.lexer import RegexLexer, words
+from pygments.token import Comment, Keyword, Name, String, Number, Punctuation, Whitespace, Literal
+
+__all__ = ['TalLexer']
+
+class TalLexer(RegexLexer):
+
+    name = 'Tal'
+    aliases = ['tal', 'uxntal']
+    filenames = ['*.tal']
+    mimetypes = ['text/x-uxntal']
+
+    instructions = [
+        'BRK', 'LIT', 'INC', 'POP', 'DUP', 'NIP', 'SWP', 'OVR', 'ROT',
+        'EQU', 'NEQ', 'GTH', 'LTH', 'JMP', 'JCN', 'JSR', 'STH',
+        'LDZ', 'STZ', 'LDR', 'STR', 'LDA', 'STA', 'DEI', 'DEO',
+        'ADD', 'SUB', 'MUL', 'DIV', 'AND', 'ORA', 'EOR', 'SFT'
+    ]
+
+    tokens = {
+        # the comment delimiters must not be adjacent to non-space characters.
+        # this means ( foo ) is a valid comment but (foo) is not. this also
+        # applies to nested comments.
+        'comment': [
+            (r'(?<!\S)\((?!\S)', Comment.Multiline, '#push'), # nested comments
+            (r'(?<!\S)\)(?!\S)', Comment.Multiline, '#pop'), # nested comments
+            (r'(\S[()]|[()]\S|[^()])+', Comment.Multiline), # comments
```

Comment:
This is going to fail on the comment

```
( ((( )
```

Please either use lookbehind/lookahead assertions, or (likely simpler) do two rules:

```python
(r'(<?!\S)\((?!\S)', Comment.Multiline, '#push'),
(r'(<?!\S)\)(?!\S)', Comment.Multiline, '#push'),
(r'[^()]+', Comment.Multiline),
(r'[()]', Comment.Multiline),
```